backend/src/schema/resolvers/shout.spec.js

```diff
@@ -19,141 +19,148 @@ const mutationUnshoutPost = gql`
       it('throws authorization error', async () => {
-        const client = new GraphQLClient(host)
-        await expect(client.request(mutationShoutPost, { id: 'p1234' })).rejects.toThrow(
-          'Not Authorised',
-        )
+        variables = { id: 'post-to-shout-id' }
+        authenticatedUser = undefined
+        await expect(mutate({ mutation: mutationShoutPost, variables })).resolves.toMatchObject({
+          errors: [{ message: 'Not Authorised!' }],
+        })
       })
     })
+    describe('authenticated', () => {
+      beforeEach(async () => {
+        currentUser = await factory.create('User', {
+          id: 'current-user-id',
+          name: 'Current User',
+          email: 'current.user@example.org',
+          password: '1234',
+        })
 
-    it('I shout a post of another user', async () => {
-      const res = await clientUser1.request(mutationShoutPost, { id: 'p12345' })
-      const expected = {
-        shout: true,
-      }
-      expect(res).toMatchObject(expected)
-
-      const { Post } = await clientUser1.request(gql`
-        query {
-          Post(id: "p12345") {
-            shoutedByCurrentUser
-          }
-        }
-      `)
-      const expected2 = {
-        shoutedByCurrentUser: true,
-      }
-      expect(Post[0]).toMatchObject(expected2)
-    })
+        postAuthor = await factory.create('User', {
+          id: 'post-author-id',
+          name: 'Post Author',
+          email: 'post.author@example.org',
+          password: '1234',
+        })
+        authenticatedUser = await currentUser.toJson()
+        await factory.create('Post', {
+          name: 'Other user post',
+          id: 'another-user-post-id',
+          author: postAuthor,
+        })
+        await factory.create('Post', {
+          name: 'current user post',
+          id: 'current-user-post-id',
+          author: currentUser,
+        })
+        variables = {}
+      })
 
-    it('I can`t shout my own post', async () => {
-      const res = await clientUser1.request(mutationShoutPost, { id: 'p1234' })
-      const expected = {
-        shout: false,
-      }
-      expect(res).toMatchObject(expected)
+      it("another user's post", async () => {
+        variables = { id: 'another-user-post-id' }
+        await expect(mutate({ mutation: mutationShoutPost, variables })).resolves.toMatchObject({
+          data: { shout: true },
+        })
+        await expect(query({ query: queryPost, variables })).resolves.toMatchObject({
+          data: { Post: [{ id: 'another-user-post-id', shoutedBy: [{ id: 'current-user-id' }] }] },
+          errors: undefined,
+        })
+      })
 
-      const { Post } = await clientUser1.request(gql`
-        query {
-          Post(id: "p1234") {
-            shoutedByCurrentUser
-          }
-        }
-      `)
-      const expected2 = {
-        shoutedByCurrentUser: false,
-      }
-      expect(Post[0]).toMatchObject(expected2)
+      it('my own post', async () => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-28T21:56:47Z" title="Saturday, September 28th 2019, 11:56:47 pm +02:00">Sep 28, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -19,141 +19,148 @@ const mutationUnshoutPost = gql`
     unshout(id: $id, type: Post)
   }
 `
-const createPostMutation = gql`
-  mutation($id: ID, $title: String!, $content: String!, $categoryIds: [ID]!) {
-    CreatePost(id: $id, title: $title, content: $content, categoryIds: $categoryIds) {
+const queryPost = gql`
+  query($id: ID!) {
+    Post(id: $id) {
       id
-      title
-      content
+      shoutedBy {
+        id
+      }
     }
   }
 `
-const createPostVariables = {
-  id: 'p1234',
-  title: 'Post Title 1234',
-  content: 'Some Post Content 1234',
-  categoryIds,
-}
-beforeEach(async () => {
-  await factory.create('User', {
-    id: 'u1',
-    email: 'test@example.org',
-    password: '1234',
-  })
-  await factory.create('User', {
-    id: 'u2',
-    email: 'test2@example.org',
-    password: '1234',
-  })
-  await instance.create('Category', {
-    id: 'cat9',
-    name: 'Democracy & Politics',
-    icon: 'university',
-  })
-  headersUser1 = await login({ email: 'test@example.org', password: '1234' })
-  headersUser2 = await login({ email: 'test2@example.org', password: '1234' })
-  clientUser1 = new GraphQLClient(host, { headers: headersUser1 })
-  clientUser2 = new GraphQLClient(host, { headers: headersUser2 })
 
-  await clientUser1.request(createPostMutation, createPostVariables)
-  await clientUser2.request(createPostMutation, {
-    id: 'p12345',
-    title: 'Post Title 12345',
-    content: 'Some Post Content 12345',
-    categoryIds,
+describe('shout and unshout posts', () => {
+  let currentUser, postAuthor
+  beforeAll(() => {
+    authenticatedUser = undefined
+    const { server } = createServer({
+      context: () => {
+        return {
+          driver,
+          neode: instance,
+          user: authenticatedUser,
+        }
+      },
+    })
+    mutate = createTestClient(server).mutate
+    query = createTestClient(server).query
+  })
+  afterEach(() => {
+    factory.cleanDatabase()
   })
-})
-
-afterEach(async () => {
-  await factory.cleanDatabase()
-})
 
-describe('shout', () => {
-  describe('shout foreign post', () => {
-    describe('unauthenticated shout', () => {
+  describe('shout', () => {
+    describe('unauthenticated', () => {
       it('throws authorization error', async () => {
-        const client = new GraphQLClient(host)
-        await expect(client.request(mutationShoutPost, { id: 'p1234' })).rejects.toThrow(
-          'Not Authorised',
-        )
+        variables = { id: 'post-to-shout-id' }
+        authenticatedUser = undefined
+        await expect(mutate({ mutation: mutationShoutPost, variables })).resolves.toMatchObject({
+          errors: [{ message: 'Not Authorised!' }],
+        })
       })
     })
+    describe('authenticated', () => {
+      beforeEach(async () => {
+        currentUser = await factory.create('User', {
+          id: 'current-user-id',
+          name: 'Current User',
+          email: 'current.user@example.org',
+          password: '1234',
+        })
 
-    it('I shout a post of another user', async () => {
-      const res = await clientUser1.request(mutationShoutPost, { id: 'p12345' })
-      const expected = {
-        shout: true,
-      }
-      expect(res).toMatchObject(expected)
-
-      const { Post } = await clientUser1.request(gql`
-        query {
-          Post(id: "p12345") {
-            shoutedByCurrentUser
-          }
-        }
-      `)
-      const expected2 = {
-        shoutedByCurrentUser: true,
-      }
-      expect(Post[0]).toMatchObject(expected2)
-    })
+        postAuthor = await factory.create('User', {
+          id: 'post-author-id',
+          name: 'Post Author',
+          email: 'post.author@example.org',
+          password: '1234',
+        })
+        authenticatedUser = await currentUser.toJson()
+        await factory.create('Post', {
+          name: 'Other user post',
+          id: 'another-user-post-id',
+          author: postAuthor,
+        })
+        await factory.create('Post', {
+          name: 'current user post',
+          id: 'current-user-post-id',
+          author: currentUser,
+        })
+        variables = {}
+      })
 
-    it('I can`t shout my own post', async () => {
-      const res = await clientUser1.request(mutationShoutPost, { id: 'p1234' })
-      const expected = {
-        shout: false,
-      }
-      expect(res).toMatchObject(expected)
+      it("another user's post", async () => {
+        variables = { id: 'another-user-post-id' }
+        await expect(mutate({ mutation: mutationShoutPost, variables })).resolves.toMatchObject({
+          data: { shout: true },
+        })
+        await expect(query({ query: queryPost, variables })).resolves.toMatchObject({
+          data: { Post: [{ id: 'another-user-post-id', shoutedBy: [{ id: 'current-user-id' }] }] },
+          errors: undefined,
+        })
+      })
 
-      const { Post } = await clientUser1.request(gql`
-        query {
-          Post(id: "p1234") {
-            shoutedByCurrentUser
-          }
-        }
-      `)
-      const expected2 = {
-        shoutedByCurrentUser: false,
-      }
-      expect(Post[0]).toMatchObject(expected2)
+      it('my own post', async () => {
```


As an improvement, you can think of the `it` as a beginning of the sentence. E.g.:

*it ... does something*

So here it would be better to write "it is allowed to shout your own post" than "it my own post" which is no proper English.